configuration/postgres/postgres.go

```diff
@@ -284,19 +286,41 @@ func (p *ConfigurationStore) handleSubscribedChange(ctx context.Context, handler
 	}
 }
 
-func (p *ConfigurationStore) connectDB(ctx context.Context) (*pgxpool.Pool, error) {
-	config, err := p.metadata.GetPgxPoolConfig()
-	if err != nil {
-		return nil, fmt.Errorf("postgres configuration store connection error : %w", err)
+func (p *ConfigurationStore) connectDB(ctx context.Context, connStr string) (*pgxpool.Pool, error) {
+	var (
+		config *pgxpool.Config
+		pool   *pgxpool.Pool
+		err    error
+	)
+	// Note: if AWS IAM enabled then must use master connection string to connect initially,
+	// otherwise connect using regular p.metadata.ConnectionString.
+	if p.metadata.UseAWSIAM {
+		config, err = p.metadata.GetPgxPoolConfig()
+		if err != nil {
+			return nil, fmt.Errorf("postgres configuration store connection error: %w", err)
+		}
+		err = awsiam.InitAWSDatabase(ctx, config, p.metadata.Timeout, connStr, p.metadata.AWSRegion, p.metadata.AWSAccessKey, p.metadata.AWSSecretKey)
```

Comment:
Ditto, must be moved to the `GetPgxPoolConfig`

Let's strive for code reuse please, rather than copy-pasting, to improve maintainability